sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InlineCTE.scala

```diff
@@ -140,26 +140,31 @@ case class InlineCTE(
       cteMap: mutable.Map[Long, CTEReferenceInfo]): LogicalPlan = {
     plan match {
       case WithCTE(child, cteDefs) =>
-        val notInlined = mutable.ArrayBuffer.empty[CTERelationDef]
-        cteDefs.foreach { cteDef =>
+        val notInlined = cteDefs.map {cteDef =>
```

Comment:
nit: squash `cteDefs.map -> Option` + `flatMap -> Seq` to `cteDefs.flatMap -> Seq`

Reply:
yeah sure done!